docs/proposals/scheduling/20240115-elastic-quota-support-job-level-oversold-management.md

```diff
@@ -0,0 +1,295 @@
+
+---
+
+# Elastic-Quota-Support-Job-Level-Oversold-Management
+
+<!-- TOC -->
+
+- [Elastic-Quota-Support-Job-Level-Oversold-Management](#elastic-quota-support-job-level-oversold-management)
+    - [Summary](#summary)
+    - [Motivation](#motivation)
+        - [Goals](#goals)
+        - [Non-goals/Future work](#non-goalsfuture-work)
+    - [Proposal](#proposal)
+        - [Key Concept/User Stories](#key-conceptuser-stories)
+        - [Implementation Details](#implementation-details)
+            - [Non-Preemptible and Preemptible](#non-preemptible-and-preemptible)
+            - [Quota Assignment](#quota-assignment)
+            - [Preemption](#preemption)
+                - [Quota And Node binding](#quota-and-node-binding)
+                - [Job Level Preemption](#job-level-preemption)
+                - [Response To Preempted](#response-to-preempted)
+                - [Preemption Node Scope](#preemption-node-scope)
+            - [Extension Point](#extension-point)
+                - [Over All](#over-all)
+                - [PreFilter](#prefilter)
+                - [PostFilter](#postfilter)
+                - [BackGroup Loop](#backgroup-loop)
+        - [API](#api)
+            - [Quota](#quota)
+            - [Pod](#pod)
+        - [Compatibility](#compatibility)
+    - [Unsolved Problems](#unsolved-problems)
+    - [Alternatives](#alternatives)
+    - [Implementation History](#implementation-history)
+    - [References](#references)
+
+<!-- /TOC -->
+
+## Summary
+This proposal provides job level elastic quota oversold mechanism for the scheduler, including resource assign and preemption.
+
+## Motivation
+In a large scale shared-cluster, some quotas may be very busy, some quotas may be idle. If the idle resources
+can be shared in some way, the resource utilization rate of the cluster will be significantly improved. Moreover, even in
+one quota, there are production jobs and develop jobs, user may expect production jobs can preempt develop jobs
+when resource is busy.
+
+If the pods belong to a job , we should also consider resource assign\preemption by job level.
+
+### Goals
+
+1. Define API to announce guaranteed and oversold job/pod. This proposal we use non-preemptible and preemptible instead of
+   guaranteed and oversold.
+
+2. Enhance the elastic quota plugin to achieve job level elastic quota oversold management.
+
+### Non-goals/Future work
+
+## Proposal
+
+### Key Concept\User Stories
```

Comment:
Reading the User Story chapter as a whole, it feels more like a Proposal rather than some specific scenarios or problems. It is recommended to split these user stories, and it is not recommended to introduce specific solutions in the user stories. Again, we should focus on one thing: clearly describing the issues we need to pay attention to.

Reply:
done